services/execsrv/observers.go

```diff
@@ -31,7 +36,49 @@ func (o *stateObserver) Name() string {
 }
 
 func (o *stateObserver) Update(state State) State {
-	log.Info().Str("exec", o.exec.ID).Msgf("No-op state observer received state update: %s", state.Status)
+	log.Info().Str("exec", o.exec.ID).Msgf("State observer received state update: %s", state.Status)
+
+	switch state.Status {
+	case types.StatusRunning:
+		log.Info().
+			Str(types.ExecIDCtxKey, o.ExecID()).
+			Str(types.ObserverCtxKey, o.Name()).
+			Msg("Handling exec state running")
+
+		exec, err := o.srv.RefreshConnectionInfo(context.Background(), o.ExecID())
+		if err != nil {
+			log.Warn().Err(err).Send()
+		}
+
+		o.exec = exec
+
+		update := db.ExecStatusUpdateParams{
+			ID:      o.ExecID(),
+			Status:  db.UnweaveExecStatus(types.StatusRunning),
+			ReadyAt: sql.NullTime{Time: time.Now(), Valid: true},
+		}
+
+		if err := db.Q.ExecStatusUpdate(context.Background(), update); err != nil {
+			log.Warn().Err(err).Send()
+		}
+
+		o.exec.Status = types.StatusRunning
+	case types.StatusTerminated:
+		log.Info().
+			Str(types.ExecIDCtxKey, o.ExecID()).
+			Str(types.ObserverCtxKey, o.Name()).
+			Msg("Handling exec termination")
+
+		if err := o.srv.Terminate(context.Background(), o.ExecID()); err != nil {
+			log.Warn().Err(err).Send()
+		}
+	case types.StatusPending,
+		types.StatusInitializing,
+		types.StatusError,
+		types.StatusFailed,
+		types.StatusSuccess,
+		types.StatusUnknown:
+	}
```

Comment:
These are todos for the future?

Reply:
Yeah, for the future, and so that we are exhaustive in this switch - there's a useful linter that will validate we have all the cases covered if we add a new status.